### ISSUE TITLE
Reject connections with conflicting destination tables

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#16173]
## Tell us about the problem you're trying to solve
The root cause of https://github.com/airbytehq/oncall/issues/473#issuecomment-1232224533 was that the user had two connections with different sources, both writing to the same destination table. This caused normalization to fail. We should have better handling for this situation.

This also manifests as failures when two connections run concurrently, since they might fight over creating/deleting ephemeral resources (snowflake stages, etc.)

## Describe the solution you’d like
Platform should prevent users from setting up multiple connections to the same connector with the same namespace + stream name+prefix.

## Describe the alternative you’ve considered or used
Currently, users just need to be careful when setting up their connections.

Some possible alternative options:
* Normalization should detect that the table already exists but has a completely mismatched schema (?). If this happens, it should fail more informatively.
* destination connectors check for this somehow (?)

**Comment by grishick** [comment]
This came up twice now in OC

**Comment by grishick** [comment]
Notes from grooming in Destinations team:
This should be solved at the platform level, possibly with a UI change that shows a warning to the user. The desired outcome is that the UI warns the user if they are trying to sync more than one stream with the same name into the same destination

**Comment by edgao** [comment]
this _keeps_ coming up, mostly unintentionally, but occasionally users intentionally do this (and we should prevent them from doing so). Maybe worth trying to build a bandaid solution in normalization? (or at least, understand why we're getting syntax errors >.>)

my minimum success story here is that normalization runs the proper `alter table` statement(s). I.e. every time the syncs run, it ping-pongs between the two schemas. This probably causes data loss - but it's recoverable, because users will still have their raw tables.

though once we get to 1 stream = 1 table... things get kind of sketchy.

**Comment by edgao** [comment]
some more examples:
* https://github.com/airbytehq/oncall/issues/1751
* https://github.com/airbytehq/oncall/issues/1768

**Comment by brunoalano** [comment]
With Airbyte Cloud, and running the destination without normalization, it's safe to use multiple sources and same destination?

**Comment by edgao** [comment]
there's a bit of nuance. There's only a problem if multiple connections are trying to write to the same tables.

For example: if you have connections from source-bing-ads and source-google-ads, both syncing the `campaigns` stream (and both connections are syncing to the same namespace and using the same stream prefix) - then that will not work.

Other than that, multiple sources + single destination is how you're intended to use airbyte. (including both cloud and oss)

**Comment by brunoalano** [comment]
@edgao But what if I've multiple source-facebook-marketing writing in the same destination, under the same namespace, that's a problem?

**Comment by edgao** [comment]
yes, that would be a problem. Airbyte connections assume that they're running in isolation. E.g. if you run a reset on one connection, that would drop/truncate the entire table.

**Comment by evantahler** [comment]
Related: https://github.com/airbytehq/oncall/issues/2330
